packages/app/src/vis-packs/nexus/utils.ts

```diff
@@ -33,17 +34,38 @@ export function isNxDataGroup(group: Group): boolean {
   return getAttributeValue(group, 'NX_class') === 'NXdata';
 }
 
+export function isNxEntry(entity: Entity): boolean {
+  return getAttributeValue(entity, 'NX_class') === 'NXentry';
+}
+
 function assertNxDataGroup(group: Group): void {
   if (!isNxDataGroup(group)) {
     throw new Error('Expected NXdata group');
   }
 }
 
 export function isNxGroup(entity: Entity): boolean {
-  return (
-    isGroup(entity) &&
-    (isNxDataGroup(entity) || !!getAttributeValue(entity, 'default'))
-  );
+  // A NxGroup is...
+  if (!isGroup(entity)) {
+    return false;
+  }
+
+  // ...a NXData group OR
+  if (isNxDataGroup(entity)) {
+    return true;
+  }
+
+  // ... a group with a 'default' attribute OR
+  // ... a NXentry group OR
+  if (getAttributeValue(entity, 'default') || isNxEntry(entity)) {
+    return true;
+  }
+
+  // ... a NXroot that has a NXentry child
+  return hasChildren(entity) &&
+    getAttributeValue(entity, 'NX_class') === 'NXroot'
+    ? entity.children.some(isNxEntry)
+    : false;
```

Comment:
If I understand `getDefaultEntityPath` correctly, it calls `isNxGroup` for every _child_ of an entity. So I don't think `isNxGroup` can ever be called with the root entity.

If you remove this, then you can shrink this function back down to a one-liner:

```ts
  return (
    isGroup(entity) &&
    (isNxDataGroup(entity) || !!getAttributeValue(entity, 'default')) || isNxEntry(entity)
  );
```

That being said, I also think `isNxEntry` is unnecessarily restrictive: we could just check for the presence of `NX_class` rather than check for `NXentry` specifically. This way we allow for `NXprocess` and the like to be selected:

```ts
  return (
    isGroup(entity) &&
    (isNxDataGroup(entity) || !!getAttributeValue(entity, 'default')) || !!getAttributeValue(entity, 'NX_class')
  );
```

Reply:
> If I understand `getDefaultEntityPath` correctly, it calls `isNxGroup` for every _child_ of an entity. 

Yes if needed.

> So I don't think `isNxGroup` can ever be called with the root entity.

In fact, it will be called for the root entity in the `VisPackChooser` to know if `NexusPack` should be used for the root entity.

So your first proposed implementation of `isNxGroup` would consider that the NexusPack should not be used for a `NXroot` without `default`.

This being said, your second implementation would work in this case and simplifies things :+1: